### PR TITLE
Add Dockerfile & build containers in CI

### PR DESCRIPTION
**Pull request by obbardc**
I wasn't sure if it would be best to build two images, one for the CLI and one for the server, or just one image with both binaries so I just built _something_ quickly ready for feedback.

### CODE REVIEW

**Comment by sjoerdsimons** [comment]
probably best to have different containers for cli and server; that would avoid having to jump  through hoops to run the server from it's container.

**Comment by obbardc** [comment]
> probably best to have different containers for cli and server; that would avoid having to jump through hoops to run the server from it's container.

Done. Added build argument which allows choice of binary to build.

File: README.md [hunk 1]
```diff
@@ -6,3 +6,36 @@ Distributed service to interact with development boards.
...
+  --help
+```
+
+
+Run `boardswarm` inside the container:
+
+```
+docker run -it --rm \
+  --port 16421:16421 \
+  --mount "type=bind,source=$(pwd)/boardswarm/share/server.conf,destination=/etc/boardswarm/server.conf" \
```
Comment (sjoerdsimons) [inline README.md:37]: why the very verbose mount line ooi

File: README.md [hunk 2]
```diff
@@ -6,3 +6,36 @@ Distributed service to interact with development boards.
...
+  ghcr.io/boardswarm/boardswarm-cli:main \
+  --help
+```
+
+
+Run `boardswarm` inside the container:
+
+```
+docker run -it --rm \
+  --port 16421:16421 \
```
Comment (sjoerdsimons) [inline README.md:36]: the default port is 6683; the 16421 is the efault port of the pdudaemon

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+RUN apt update && \
+    apt install --yes \
+      libudev1 \
+      libssl3 && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /usr/src/boardswarm/target/release/$BINARY /usr/local/bin/$BINARY
+
+ENTRYPOINT ["/usr/local/bin/$BINARY"]
```
Comment (sjoerdsimons) [inline Dockerfile:44]: for boardswarm you probably want the default entrypoint to have a default configuration path already for ease of use

File: README.md [hunk 4]
```diff
@@ -6,3 +6,36 @@ Distributed service to interact with development boards.
...
+  --mount "type=bind,source=$(pwd)/boardswarm/share/server.conf,destination=/etc/boardswarm/server.conf" \
+  --entrypoint boardswarm \
```
Comment (sjoerdsimons) [inline README.md:38]: why is the entrypoint being overriden rather then using the default form the container

File: README.md (hunk 1 above)
Comment (obbardc) [reply README.md:37]: no reason; changed to --volume now

File: README.md (hunk 2 above)
Comment (obbardc) [reply README.md:36]: oops, solved

File: Dockerfile (hunk 3 above)
Comment (obbardc) [reply Dockerfile]: done had to separate the dockerfiles.

**Pull request merged**